drivers/entropy/entropy_npcx_drbg.c

```diff
@@ -0,0 +1,246 @@
+			ncl_ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int entropy_npcx_drbg_suspend(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	void *ctx = data->handle;
+
+	return entropy_npcx_drbg_enable_drbg_power(ctx, false);
+}
+
+static int entropy_npcx_drbg_resume(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	void *ctx = data->handle;
+
+	return entropy_npcx_drbg_enable_drbg_power(ctx, true);
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static int entropy_npcx_drbg_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	enum ncl_status ncl_ret;
+	void *ctx = data->handle;
+	int ret = 0;
+
+	k_sem_take(&data->sem_lock, K_FOREVER);
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, true);
+	if (ret != 0) {
+		goto err_exit;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->generate(ctx, NULL, 0, buf, len);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to generate: err 0x%02x", ncl_ret);
+		ret = -EIO;
+		goto err_exit;
+	}
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, false);
+	if (ret != 0) {
+		goto err_exit;
+	}
+
+err_exit:
+	k_sem_give(&data->sem_lock);
+
+	return ret;
+}
+
+static int entropy_npcx_drbg_init(const struct device *dev)
+{
+	struct entropy_npcx_drbg_dev_data *const data = dev->data;
+	uint32_t handle_size_required;
+	enum ncl_status ncl_ret;
+	void *ctx = data->handle;
+	int ret;
+
+	handle_size_required = NPCX_NCL_DRBG->get_context_size();
+	if (handle_size_required != NPCX_DRBG_HANDLE_SIZE) {
+		LOG_ERR("Unexpected NCL DRBG context_size = %d", handle_size_required);
+		return -ENOSR;
+	}
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, true);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = entropy_npcx_drbg_enable_drbg_power(ctx, true);
+	ncl_ret = NPCX_NCL_DRBG->power(ctx, true);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->init_context(ctx);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to init ctx: err 0x%02x", ncl_ret);
+		return -EIO;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->init(ctx, false);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to init: err 0x%02x", ncl_ret);
+		return -EIO;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->config(ctx, NPCX_DRBG_RESEED_INTERVAL, false);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to config: err 0x%02x", ncl_ret);
+		return -EIO;
+	}
+
+	ncl_ret = NPCX_NCL_DRBG->instantiate(ctx, NPCX_DRBG_SECURITY_STRENGTH, NULL, 0);
+	if (ncl_ret != NCL_STATUS_OK) {
+		LOG_ERR("Fail to config: err 0x%02x", ncl_ret);
+		return -EIO;
+	}
+
+	ret = entropy_npcx_drbg_enable_sha_power(ctx, false);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Locking semaphore initialized to 1 (unlocked) */
+	k_sem_init(&data->sem_lock, 1, 1);
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int entropy_npcx_drbg_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	int res = 0;
```

Comment:
no need to initialize this one, you may also drop it entirely and just

```c

	switch (action) {
	case PM_DEVICE_ACTION_SUSPEND:
		return entropy_npcx_drbg_suspend(dev);
```

etc

Reply:
Done.